src/engine/UI/menu/MenuOption.cpp

```diff
@@ -94,9 +94,30 @@ void CycleOption::Cycle() {
 
 size_t CycleOption::CurrentIndex() const { return m_currentOption; }
 
-InputOption::InputOption(const std::string &name, float order,
-                         const std::string &placeholder)
-    : MenuOption(name, order, ""), m_placeholderText(placeholder) {}
+InputOption::InputOption(
+    const std::string &name, float order,
+    events::EventSystem<events::KeyPressEventArgs> &eventHandler,
+    const std::string &placeholder)
+    : MenuOption(name, order, ""), m_placeholderText(placeholder),
+      m_keyPressEventHandler(eventHandler) {
```

Comment:
We should add a string value argument to the constructor so the initial value can be set instead of defaulting to an empty string.

Reply:
Sure, perhaps we can use a default-value parameter/argument.